plugins/woocommerce-admin/client/marketplace/components/header-account/header-account.tsx

```diff
@@ -1,32 +1,137 @@
 /**
  * External dependencies
  */
+import { DropdownMenu, MenuGroup, MenuItem } from '@wordpress/components';
+import { Icon, commentAuthorAvatar, external, linkOff } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
+import './header-account.scss';
+import { getAdminSetting } from '../../../utils/admin-settings';
 
-export default function HeaderAccount() {
-	return (
-		<a
-			className="woocommerce-marketplace__header-account"
-			href="https://woocommerce.com/my-account/"
-		>
-			<svg
-				width="24"
-				height="24"
-				viewBox="0 0 24 24"
-				fill="none"
-				xmlns="http://www.w3.org/2000/svg"
+export default function HeaderAccount(): JSX.Element {
+	const wccomSettings = getAdminSetting( 'wccomHelper', {} );
+	const isConnected = wccomSettings?.isConnected ?? false;
+	const connectionURL = wccomSettings?.connectURL;
+	const userEmail = wccomSettings?.userEmail;
+
+	const userData = getAdminSetting( 'currentUserData' );
+	const avatarURL = userData?.avatar_urls?.[ '48' ] ?? commentAuthorAvatar;
```

Comment:
I wonder if we're getting this from the right place? This is returning a generic grey avatar for me. But the old skool PHP connect panel is [calling](https://github.com/woocommerce/woocommerce/blob/ef015f740f89c35cebf34bc0c39a971be6723b68/plugins/woocommerce/includes/admin/helper/views/html-section-account.php#L9) `get_avatar( $auth_user_data['email'], 48 )`, which shows my usual avatar.

<img width="300" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/0ad34174-3f80-49c8-8853-705fd318cf72">

Reply:
Ah! That function pulls the gravatar directly. I was pulling an existing bit of data that is based on the admin email address. In our case, that email is the local environment email which doesn't have a gravatar! Fixed this to now pull that directly using the WCCOM account email address.